tests/redirect.rs

```diff
@@ -92,3 +107,114 @@ fn test_redirect() {
     assert!(called_redirect == called_redirect3);
     assert!(called_redirect == called_about);
 }
+
+#[test]
+#[with_mock_server]
+/// Simulate a load test which includes a page with a redirect to another domain
+/// (which in this case is a second mock server running on a different path).
+/// all redirects are correctly followed.
+fn test_domain_redirect() {
+    let mock_index = mock(GET, INDEX_PATH).return_status(200).create();
+    let mock_about = mock(GET, ABOUT_PATH).return_status(200).create();
+    let alternate_domain = &mockito::server_url();
+    let mock_redirect = mock(GET, REDIRECT_PATH)
+        // Moved Permanently
+        .return_status(301)
+        .return_header(
+            "Location",
+            format!("{}{}", alternate_domain, INDEX_PATH).as_str(),
+        )
+        .create();
+    let mock_index_alt = mockito::mock("GET", INDEX_PATH)
+        .with_status(200)
+        .expect_at_least(1)
+        .create();
+    let mock_about_alt = mockito::mock("GET", ABOUT_PATH)
+        .with_status(200)
+        .expect(0)
+        .create();
+
+    crate::GooseAttack::initialize_with_config(common::build_configuration())
+        .setup()
+        .register_taskset(
+            taskset!("LoadTest")
+                // First load redirect, takes this request only to another domain.
+                .register_task(task!(get_domain_redirect).set_on_start())
+                // Load index directly.
+                .register_task(task!(get_index))
+                // Load about directly, always on original domain.
+                .register_task(task!(get_about)),
+        )
+        .execute();
+
+    let called_index = mock_index.times_called();
+    let called_about = mock_about.times_called();
+    let called_redirect = mock_redirect.times_called();
+
+    // Confirm that we load the index, about and redirect pages on the orginal domain.
+    assert_ne!(called_index, 0);
+    assert_ne!(called_about, 0);
+    assert_ne!(called_redirect, 0);
+
+    // Confirm that the redirect sends us to the second domain (mocked using a
+    // server on a different port).
+    mock_index_alt.assert();
+
+    // Confirm the we never loaded the about page on the second domain.
+    mock_about_alt.assert();
+}
+
+#[test]
+#[with_mock_server]
+fn test_sticky_domain_redirect() {
```

Comment:
@LionsAd This test confirms that with `--sticky-follow` enabled redirects ARE sticky